lib/matplotlib/tests/test_colors.py

```diff
@@ -1728,3 +1728,37 @@ def test_colorizer_vmin_vmax():
     assert ca.vmax == 3.0
     assert ca.norm.vmin == 1.0
     assert ca.norm.vmax == 3.0
+
+
+def test_lsc_from_list_color_alpha_tuple():
+    """
+    GitHub issue #29042: A bug in 'from_list' causes an error
+    when passing a tuple (str, float) where the string is a
+    color name or grayscale value and float is an alpha value.
+    """
+    valid_colors = [("red", 0.3), ("0.42", 0.1), "green"]
+    mcolors.LinearSegmentedColormap.from_list("testcolormap", valid_colors)
+
+
+def test_lsc_from_list_mixed_tuples():
```

Comment:
Might be worth looking at `pytest.mark.parameterize` (look for example in the tests) which lets you make a list of input and then verify they all behave the same.  It is a bit of extra machinery, but it would let you merge the two cases in this test and the next test into one test

```python
@pytest.mark.parameterize('inp,e_type',
[
(..., TypeError),
(..., TypeError),
(..., ValueError)
]
def test_lsc_failures(inp, e_type):
    with ptest.raises(e_type):
        mcolors.LinearSegmentedColormap.from_list("testcolormap2", inp)
```